src/app/components/algorithms/implementation-view/implementation-nisq-analyzer-qpu-selection/implementation-nisq-analyzer-qpu-selection.component.ts

```diff
@@ -268,11 +344,209 @@ export class ImplementationNisqAnalyzerQpuSelectionComponent implements OnInit {
       });
   }
 
+  prioritize(): void {
+    this.rankings = [];
+    this.utilService
+      .createDialog(
+        ImplementationNisqAnalyzerQpuSelectionPrioritizationDialogComponent,
+        {
+          title: 'Prioritize QPU-Selection-Analysis',
+        }
+      )
+      .afterClosed()
+      .subscribe((dialogResult) => {
+        let totalSum = 0;
+        if (dialogResult) {
+          this.loadingMCDAJob = true;
+          this.prioritizationJob = undefined;
+          this.prioritizationJobReady = false;
+          // calculate SMART with new assigned points
+          dialogResult.criteriaAndValues.forEach((obj) => {
+            totalSum += obj.points;
+          });
+          dialogResult.criteriaAndValues.forEach((obj) => {
+            if (obj.points !== 0) {
+              obj.weight = obj.points / totalSum;
+            } else {
+              obj.weight = 0;
+            }
+          });
+          let numberOfCriterion = 0;
+          dialogResult.criteriaAndValues.forEach((obj) => {
```

Comment:
What about enabling the backend to support storing a list of all the values? This is really easy to do in the backend and makes the UI much easier here.

Reply:
yeah instead of make a PUT on each criteria, adapting the hole list at once could be more simple in this case.. I will make an issue for the backend